crates/subspace-farmer/src/farming.rs

```diff
@@ -135,11 +185,25 @@ async fn subscribe_to_slot_info<T: RpcClient>(
 
             move || {
                 let local_challenge = derive_local_challenge(slot_info.global_challenge, &identity);
-                match commitments.find_by_range(
-                    local_challenge.derive_target(),
-                    slot_info.solution_range,
-                    slot_info.salt,
-                ) {
+                // Try to first find a block authoring solution, then if not found try to find a vote
```

Comment:
How heavy is a call to `find_by_range`()?  I think you could save almost everyone (except winner and voters) one call if you first check for `voting_solution_range` since block authoring `solution_range` is contained within it.
Right now the winner only does 1 call and everyone else 2. If you first check voting range, the winner and voters will do 2 calls and everyone else 1.

Reply:
If we just check for `voting_solution_range` we'll potentially find solutions that are only sufficient for voting, but not block authoring, hence block authoring frequency will decrease even if we have corresponding commitments.

The call itself isn't very heavy, but it is a database access, so it isn't free for sure.

I think we can optimize it a bit though, will submit a PR later.